src/Agent.Worker/Release/Artifacts/FileShareArtifact.cs

```diff
@@ -35,10 +35,18 @@ public class FileShareArtifact
             {
                 foreach (var filePath in filePaths)
                 {
-                    var filePathRelativeToDrop = filePath.Replace(dropLocation, string.Empty).Trim(trimChars);
-                    using (var fileReader = fileSystemManager.GetFileReader(filePath))
+                    var fullPath = Path.GetFullPath(filePath);
+                    if (File.Exists(fullPath))
                     {
-                        await fileSystemManager.WriteStreamToFile(fileReader.BaseStream, Path.Combine(localFolderPath, filePathRelativeToDrop));
+                        var filePathRelativeToDrop = filePath.Replace(dropLocation, string.Empty).Trim(trimChars);
+                        using (var fileReader = fileSystemManager.GetFileReader(filePath))
```

Comment:
can you please change these two vars to actual type. I know you didn't make this change. But unless if type is not explicit on the right hand side we try to declare the type on the left.